core/templates/dev/head/components/SearchBarDirective.js

```diff
@@ -151,11 +164,49 @@ oppia.directive('searchBar', [function() {
             }
 
             _updateSelectionDetails('languageCodes');
+
+            if (Boolean(urlService.getUrlParams().q)) {
+              _parseURL();
+            }
             _onSearchQueryChangeExec();
 
             _searchBarFullyLoaded = true;
           }
         );
+
+        var _parseURL = function() {
+          var urlSearch = decodeURI($window.location.search);
+          var query = urlSearch.replace('?q=', '');
```

Comment:
Hm. Immediately before you call _parseURL(), you get the query using urlService.getUrlParams().q. Why don't you just pass that in, and have the rest of this method not depend on $window at all? That seems somewhat cleaner.